components/NavBar.js

```diff
@@ -209,60 +230,57 @@ const NavBar = () => {
   }
 
   const handleScroll = () => {
-    const scroll = window.pageYOffset || document.documentElement.scrollTop;
-    if (scroll <= 0) {
-      setVisibility('visible');
-      setOpacity('1');
-    } else if (scroll > lastScroll) {
-      setVisibility('hidden');
-      setOpacity('0');
-    } else {
-      setVisibility('visible');
-      setOpacity('1');
+    var lastScroll = 0;
+    return () => {
+      const scroll = window.pageYOffset || document.documentElement.scrollTop;
+      if (scroll <= 0) {
+        setVisibility('visible');
+        setOpacity('1');
+      } else if (scroll > lastScroll) {
+        setVisibility('hidden');
+        setOpacity('0');
+      } else {
+        setVisibility('visible');
+        setOpacity('1');
+      }
+      lastScroll = scroll;
```

Comment:
I see that I had made scroll a global variable before and that wasn't behaving as we want it. But I'm a bit confused about the line `lastScroll = scroll;` is doing: doesn't `lastScroll` get set to 0 at first anyways when the `handleScroll` function is called?

Reply:
I'm not entirely sure either, but I don't think the `var lastScroll = 0` line is being executed every time the scroll event fires. How it works in my head is:
1. scroll event fires and checks what `handleScroll` is referring to
2. `handleScroll` initializes `lastScroll` as `0` the _first_ time, and returns an anonymous function
3.  every time the scroll event fires, it executes the anonymous function, which sets `lastScroll = scroll`